### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by iniakponode**
This pull request introduces several major improvements to the High School Management System, focusing on activity management, user experience enhancements, and automated testing. The most important changes are the addition of participant management features (including unregistering), UI updates to display and manage participants, and a comprehensive test suite to ensure API reliability.

**Backend enhancements:**

* Added the ability to unregister students from activities via a new `DELETE /activities/{activity_name}/unregister` endpoint, including validation to prevent unregistering non-participants or from non-existent activities.
* Enhanced the signup endpoint to prevent duplicate registrations by checking if a student is already signed up for an activity.
* Expanded the initial activities data to include more clubs and teams, each with sample participants.

**Frontend improvements:**

* Updated the activities page UI to display a list of current participants for each activity, with the ability to remove participants directly from the interface.
* Implemented event delegation and logic to handle participant removal from the UI, including confirmation prompts and automatic list refreshes after changes. [[1]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R108-R153) [[2]](diffhunk://#diff-7146e1c8a61ac4e715182863a67d7a5a3178a73033e8344cd24f1c67787236e3R97-R99)
* Improved the visual styling of the participants list and delete buttons for better usability and clarity.

**Testing:**

* Added a comprehensive test suite (`tests/test_app.py`) covering all API endpoints, including edge cases for signup and unregister functionality, and integration scenarios for typical user workflows.
* Created a `tests` package to organize and initialize tests.

These changes collectively make the activity management system more robust, user-friendly, and maintainable.

### CODE REVIEW

File: tests/test_app.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+        response = client.post(
+            "/activities/Art%20Studio%20Workshop/signup?email=newartist@mergington.edu"
+        )
+        assert response.status_code == 200
+        
+        # Verify participant was added
+        activities_response = client.get("/activities")
+        activities_data = activities_response.json()
+        assert "newartist@mergington.edu" in activities_data["Art Studio Workshop"]["participants"]
+
```
[the remark this answers is not in the document]
Comment (iniakponode) [reply tests/test_app.py:190]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/iniakponode/skills-getting-started-with-github-copilot/pull/2#discussion_r2540203044)

[the remark this answers is not in the document]
Comment (iniakponode) [reply tests/test_app.py:190]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/iniakponode/skills-getting-started-with-github-copilot/pull/2#discussion_r2540203044)

[the remark this answers is not in the document]
Comment (iniakponode) [reply tests/test_app.py:190]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/iniakponode/skills-getting-started-with-github-copilot/pull/2#discussion_r2540203041)

File: tests/test_app.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+        }
+    }
+    
+    # Reset to original state
+    activities.clear()
+    activities.update(original_activities)
+    yield
+    # Clean up after test
+    activities.clear()
+    activities.update(original_activities)
```
[the remark this answers is not in the document]
Comment (iniakponode) [reply tests/test_app.py:83]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/iniakponode/skills-getting-started-with-github-copilot/pull/2#discussion_r2540203035)

**Comment by iniakponode** [comment]
@copilot open a new pull request to apply changes based on the comments in [this thread](https://github.com/iniakponode/skills-getting-started-with-github-copilot/pull/2#pullrequestreview-3480417709)
